py/mphal.h

```diff
@@ -55,6 +55,22 @@ void mp_hal_stdout_tx_strn(const char *str, size_t len);
 void mp_hal_stdout_tx_strn_cooked(const char *str, size_t len);
 #endif
 
+#ifndef mp_hal_stdin_rx_buff_poll
+uintptr_t mp_hal_stdin_rx_buff_poll(uintptr_t poll_flags);
```

Comment:
These declarations don't really belong here, because they are not part of the MP HAL API.  That API is the set of functions a port needs to provide to interface with the rest of the runtime.  The functions added here are not that, instead they are helpers that a port can choose to use to implement the required ones (ie implement `mp_hal_stdin_rx_chr()`.